app/models/test_result.rb

```diff
@@ -30,7 +30,7 @@ class TestResult < ApplicationRecord
   after_save :update_cached_fields!
   after_destroy :update_cached_fields!
 
-  scope :latest, -> { joins(with_latest) }
+  scope :latest, ->(j = 'JOIN') { joins(with_latest(j)) }
```

Comment:
Making the JOIN parametrized here so we can do a `LEFT OUTER JOIN` in some cases, but keeping the default as it was.